docs/tutorials/05-external-jsonschema.md

```diff
@@ -0,0 +1,311 @@
+meta:
+  schemaVersion: v2
+
+component:
+name: github.com/gardener/landscaper/external-jsonschema/echo-server
+version: v0.1.0
+
+provider: internal
+
+repositoryContexts:
+- type: ociRegistry
+  baseUrl: eu.gcr.io/gardener-project/landscaper/tutorials/components
+
+sources: []
+componentReferences:
+- name: definitions
+  componentName: github.com/gardener/landscaper/external-jsonschema/definitions
+  version: v0.1.0
+
+resources:
+- type: ociImage
+  name: echo-server-image
+  version: v0.2.3
+  relation: external
+  access:
+  type: ociRegistry
+  imageReference: hashicorp/http-echo:0.2.3
+```
+
+</details>
+
+When the component reference is added to the echo-server's component descriptor, the resource import can be added to the blueprint.
+The import also contains a valid jsonschema but uses the `$ref` attribute with a custom landscaper protocol implementation.
+That custom protocol specifies that the jsonschema should be fetched from the resource `resources-definition` of the referenced component `definitions`.
+
+For detailed explanation about available reference methods see the [blueprint jsonschema documentation](../usage/JSONSchema.md)
+
+```yaml
+imports:
+- name: resources
+  schema:
+    $ref: "cd://componentReferences/definitions/resources/resources-definition"
+```
+
+The blueprint is defined, so it can be added to the component descriptor using the component-cli and then uploaded.
+See the details section for concrete resource definitions and cli calls.
+
+<details>
+
+```yaml
+# ./docs/tutorials/resources/external-jsonschema/echo-server/blueprint.yaml
+apiVersion: landscaper.gardener.cloud/v1alpha1
+kind: Blueprint
+
+imports:
+- name: cluster
+  targetType: landscaper.gardener.cloud/kubernetes-cluster
+- name: ingressClass
+  schema:
+    type: string
+- name: resources
+  schema:
+    $ref: "cd://componentReferences/definitions/resources/resources-definition"
+
+deployExecutions:
+- name: default
+  type: GoTemplate
+  file: /defaultDeployExecution.yaml
+```
+
+```yaml
+# ./docs/tutorials/resources/external-jsonschema/echo-server/blueprint-resource.yaml
+---
+type: blueprint
+name: echo-server-blueprint
+version: v0.1.0
+relation: local
+input:
+  type: "dir"
+  path: "./blueprint"
+  compress: true
+  mediaType: "application/vnd.gardener.landscaper.blueprint.v1+tar+gzip"
+...
+```
+
+```
+component-cli ca resources add ./docs/tutorials/resources/external-jsonschema/echo-server -r ./docs/tutorials/resources/external-jsonschema/echo-server/blueprint-resource.yaml -v 5
+```
+
+```
+component-cli ca remote push ./docs/tutorials/resources/external-jsonschema/echo-server
+```
+
+</details>
+
+### Installation
+
+The components with their jsonschema and blueprint artifacts are now uploaded and can be deployed.
+
+To deploy the component using Landscaper, first the target import and the data imports have to be defined and applied to the Kubernetes cluster.
```

Comment:
```suggestion
In a first step, the target import and the data imports have to be defined and applied to the Kubernetes cluster so that Landscaper can pick them up.
```